txnsync/encodedgroupstypes.go

```diff
@@ -0,0 +1,357 @@
+
+	TotalTransactionsCount uint64 `codec:"ttc"`
+	TransactionGroupCount  uint64 `codec:"tgc"`
+	TransactionGroupSizes  []byte `codec:"tgs,allocbound=maxEncodedTransactionGroup"`
+
+	encodedSignedTxns
+}
+
+type encodedSignedTxns struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Sig        []byte  `codec:"sig,allocbound=maxSignatureBytes"`
+	BitmaskSig bitmask `codec:"sigbm"`
+
+	encodedMsigs
+	encodedLsigs
+
+	AuthAddr        []byte  `codec:"sgnr,allocbound=maxAddressBytes"`
+	BitmaskAuthAddr bitmask `codec:"sgnrbm"`
+
+	encodedTxns
+}
+
+type encodedMsigs struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Version          []byte  `codec:"msigv,allocbound=maxEncodedTransactionGroup"`
+	BitmaskVersion   bitmask `codec:"msigvbm"`
+	Threshold        []byte  `codec:"msigthr,allocbound=maxEncodedTransactionGroup"`
+	BitmaskThreshold bitmask `codec:"msigthrbm"`
+	// splitting subsigs further make the code much more complicated / does not give gains
+	Subsigs        [][]crypto.MultisigSubsig `codec:"subsig,allocbound=maxEncodedTransactionGroup,allocbound=crypto.MaxMultisig"`
+	BitmaskSubsigs bitmask                   `codec:"subsigsbm"`
+}
+
+type encodedLsigs struct {
+	_struct          struct{}   `codec:",omitempty,omitemptyarray"`
+	Logic            [][]byte   `codec:"lsigl,allocbound=maxEncodedTransactionGroup,allocbound=config.MaxLogicSigMaxSize"`
+	BitmaskLogic     bitmask    `codec:"lsiglbm"`
+	LogicArgs        [][][]byte `codec:"lsigarg,allocbound=maxEncodedTransactionGroup,allocbound=transactions.EvalMaxArgs,allocbound=config.MaxLogicSigMaxSize"`
+	BitmaskLogicArgs bitmask    `codec:"lsigargbm"`
+}
+
+type encodedTxns struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	TxType        []byte  `codec:"type,allocbound=maxEncodedTransactionGroup"`
+	BitmaskTxType bitmask `codec:"typebm"`
+	TxTypeOffset  byte    `codec:"typeo"`
+
+	encodedTxnHeaders
+	encodedKeyregTxnFields
+	encodedPaymentTxnFields
+	encodedAssetConfigTxnFields
+	encodedAssetTransferTxnFields
+	encodedAssetFreezeTxnFields
+	encodedApplicationCallTxnFields
+	encodedCompactCertTxnFields
+}
+
+type encodedTxnHeaders struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Sender             []byte              `codec:"snd,allocbound=maxAddressBytes"`
+	BitmaskSender      bitmask             `codec:"sndbm"`
+	Fee                []basics.MicroAlgos `codec:"fee,allocbound=maxEncodedTransactionGroup"`
+	BitmaskFee         bitmask             `codec:"feebm"`
+	FirstValid         []basics.Round      `codec:"fv,allocbound=maxEncodedTransactionGroup"`
+	BitmaskFirstValid  bitmask             `codec:"fvbm"`
+	LastValid          []basics.Round      `codec:"lv,allocbound=maxEncodedTransactionGroup"`
+	BitmaskLastValid   bitmask             `codec:"lvbm"`
+	Note               [][]byte            `codec:"note,allocbound=maxEncodedTransactionGroup"` // TODO whats the correct allocbound?
```

Comment:
The allocbound definition should be 
 `codec:"note,allocbound=maxEncodedTransactionGroup,allocbound=config.MaxTxnNoteBytes"`